### PR TITLE
userapi: Moves user APIs from ecctl

### PR DESCRIPTION
**Pull request by karencfv**
## Description
Moves the last of the APIs (userauthadminapi, userauthapi and userapi) from ecctl to cloud-sdk-go 🎉 

There has been some refactoring in the tests to use request assertion and testify.

Note for reviewer: As this is a lot of code I have separated the PR into commits that hopefully make sense

## How Has This Been Tested?
unit tests

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: pkg/api/userapi/authapi/adminapi/deletekey_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		{
+			name: "succeeds",
+			args: args{params: DeleteKeyParams{
+				API: api.NewMock(
+					mock.New200ResponseAssertion(
+						&mock.RequestAssertion{
+							Header: api.DefaultWriteMockHeaders,
+							Method: "DELETE",
+							Host:   api.DefaultMockHost,
+							Path:   "/api/v1/regions/users/someid/auth/keys/somekey",
```
Comment (marclop) [inline pkg/api/userapi/authapi/adminapi/deletekey_test.go:72]: I was mentioning it to you in Slack, something's broken on the `api.CloudClientRuntime` which makes this URL no good. I've identified where it's broken and have a branch to fix it.

Comment (karencfv) [reply pkg/api/userapi/authapi/adminapi/deletekey_test.go:72]: ah right! I'll wait for that to be merged then :)

Comment (marclop) [reply pkg/api/userapi/authapi/adminapi/deletekey_test.go:72]: #156 should fix it, if you rebase, you'll see a lot of tests failing here 🙂 🙈

File: pkg/api/userapi/enable_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+}`
+
+	type args struct {
+		params EnableParams
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *models.User
+		wantErr bool
```
Comment (marclop) [inline pkg/api/userapi/enable_test.go:54]: We can remove this `wantErr`

File: pkg/util/email/email.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// Validate validates that a string is in a valid email format
+func Validate(email string) error {
+	emailRegexp := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
+
+	if len(email) > 254 {
+		return errors.New("an email address must not exceed 254 characters")
+	}
+
+	if !emailRegexp.MatchString(email) {
+		return errors.Errorf("%v is not a valid email address format", email)
```
Comment (marclop) [inline pkg/util/email/email.go:36]: We can actually change this to use `fmt.Errorf`. Not sure why (I think I did) "github.com/pkg/errors" would be preferred over of `fmt`

**Pull request merged**